pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -586,6 +613,12 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 		return err
 	}
 
+	// Find all ImageContentSourcePolicy objects
+	icspRules, err := ctrl.icspLister.List(labels.Everything())
+	if err != nil {
```

Comment:
if this has been added in 4.2.master and 4.1 doesn't have it, we need to tolerate a notfound error or 4.1->4.2 cluster upgrade fails

Reply:
AFAICS `icspLister.List` can never fail just because an object is missing (though maybe something creating the `icspLister` might fail, I haven’t read all of the code); only `icspLister.Get` can return a `StatusReasonNotFound` error.

Anyway, added a check to make the behavior explicit.